gc/gc-iceberg/src/main/java/org/projectnessie/gc/iceberg/ComputeExpiredFilesUDF.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.gc.iceberg;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.iceberg.GcMetadataUtil;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableMetadataParser;
+import org.apache.iceberg.io.FileIO;
+import org.apache.spark.sql.api.java.UDF1;
+
+class ComputeExpiredFilesUDF implements UDF1<String, List<String>> {
+
+  private final FileIO io;
+  private final Set<String> expiredManifests;
+
+  public ComputeExpiredFilesUDF(FileIO io, Set<String> expiredManifests) {
+    this.io = io;
+    this.expiredManifests = expiredManifests;
+  }
+
+  @Override
+  public List<String> call(String metadataLocation) {
```

Comment:
This function collects all data files from all manifests to delete those. Can't a data file be in an expired **and** in a live snapshot's manifest?

Reply:
data files are not shared between manifests. Only manifests are reused between snapshots.
So, already this udf taken an arguments of `expiredManifests`. It just reads those manifests to get the data files. 

 The another problem with Iceberg interface is to read data files, manifestPath is not enough. It needs GenericManifestFile object.
On previous read, I can't store GenericManifestFile object. (bulky object and seriaization fails due to avro schema in that object with null type). So, for expired contents, I am reading table metadata and manifestList twice.

Reply:
hmm, I stand corrected. 
There is one exception to my above statement. **rewrite_manifests** :(